Fix error thrown by gsw_p_from_z.m in TEOS-10 toolkit

tayden opened the pull request:
gsw_p_from_z is throwing `Integers can only be combined with integers of the same class, or scalar doubles` when passing the input depth or pressure vector created in binRBR.m. The binRBR.m function creates int64 vectors by default. This pull request resolves the gsw error by explicitly casting the binned int64 pressure and depths vectors to type double.

tayden committed:
Remove trailing whitespace

tayden committed:
Revert "Explicitly cast binned depth values to double"

This reverts commit 9705567278498c1048f19ead956ad757d38177a1.

tayden committed:
Explicitly cast binned depth values to double

The TEOS-10 toolkit sometimes throws erros due to mismatched int64 and double vectors being multiplied (line 154 in gsw_p_from_z.m). Explicitly casting the binned depth vector to double suppresses these errors

mhalvers commented:
This is a bit strange.  This means that the pressure time series is of class int64, which should not be the case.  Pressure needs to have decimal values (i.e. class single or double).  There is nothing in binRBR.m that would coerce the data into int64.   The int64 must be coming from elsewhere.

tayden commented:
It's not that that the pressure series is int64, but when the binWidth is an integer, an array is created of integers since it uses the binWidth as the step size and initial value of a new array. This array is created in [line 46](https://github.com/mhalvers/RBRproc/blob/master/binRBR.m#L46) and [line 51](https://github.com/mhalvers/RBRproc/blob/master/binRBR.m#L51).

The error doesn't occur if the function is passed a double as the binWidth.

i.e.
```
% error thrown
binRBR(inData, "depth", 1)

% no problem
binRBR(inData, "depth", 1.0)
```

Maybe it's better not to cast the binned pressure/depth vectors to double, but to instead cast the binWidth to double incase someone calls the function with an integer as a binWidth instead of a double?

tayden committed:
Explicitly cast binWidth parameter to double

Cast binWidth function parameter to double incase it is an integer. This prevents errors from the gsw_p_from_z.m function which complains about mismatched datatypes.

tayden committed:
Merge branch 'master' of github.com:mhalvers/RBRproc into feature/fix-gsw-o-from-z-error

tayden committed:
Don't cast depth binCenter array to double

tayden commented:
This actually only occurs for depth binning, where out.Pressure is calculated from the depth using the gsw_p_from_z function

mhalvers commented:
Something still seems strange to me.  The default class for numbers is double, regardless of whether you use a decimal point.  `class(1)` and `class(1.0)` both return `double`.

tayden commented:
I agree it seems odd. Good to know that's the default. I'll keep digging around.

mhalvers commented:
Are you calling `binRBR.m` in the same way that you wrote above?  That would definitely cause an error (but a different one).

tayden commented:
I made a mistake there, I'm calling it with the dataset as the first argument, then "depth", and binWidth. I've edited the above comment

tayden commented:
I'm starting to thinking this is maybe an issue with the Python-Matlab engine I'm using. I'm calling the function via Python, and I think it may not be converting my binWidth (which is a Python integer type) to a double the same way that Matlab does by default

mhalvers commented:
I think then it makes sense for you to fix things on the python side.  If I merge all of your changes are we back to the beginning?

tayden commented:
I'd leave it for now and I'll make sure. If they don't need to be made I'll just blow away the changes
